packages/cli/commands/import/workflow.ts

```diff
@@ -93,41 +115,94 @@ export class ImportWorkflowsCommand extends Command {
 
-				if (!Array.isArray(fileContents)) {
-					throw new Error(`File does not seem to contain workflows.`);
+				if (!Array.isArray(workflows)) {
+					throw new Error(
+						'File does not seem to contain workflows. Make sure the workflows are contained in an array.',
+					);
 				}
 
-				for (i = 0; i < fileContents.length; i++) {
-					if (credentialsEntities.length > 0) {
-						// eslint-disable-next-line
-						fileContents[i].nodes.forEach((node: INode) => {
-							this.transformCredentials(node, credentialsEntities);
+				for (i = 0; i < workflows.length; i++) {
+					if (credentials.length > 0) {
+						workflows[i].nodes.forEach((node: INode) => {
+							this.transformCredentials(node, credentials);
 						});
 					}
-					// eslint-disable-next-line no-await-in-loop, @typescript-eslint/no-non-null-assertion
-					await Db.collections.Workflow!.save(fileContents[i]);
+					await this.storeWorkflow(workflows[i], user);
 				}
 			}
 
 			console.info(`Successfully imported ${i} ${i === 1 ? 'workflow.' : 'workflows.'}`);
-			process.exit(0);
+			process.exit();
 		} catch (error) {
-			console.error('An error occurred while exporting workflows. See log messages for details.');
-			// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
-			logger.error(error.message);
+			console.error('An error occurred while importing workflows. See log messages for details.');
+			if (error instanceof Error) logger.error(error.message);
 			this.exit(1);
 		}
 	}
+
+	private async initOwnerWorkflowRole() {
+		const ownerWorkflowRole = await Db.collections.Role!.findOne({
+			where: { name: 'owner', scope: 'workflow' },
+		});
+
+		if (!ownerWorkflowRole) {
+			throw new Error(`Owner workflow role not found. ${FIX_INSTRUCTION}`);
+		}
+
+		this.ownerWorkflowRole = ownerWorkflowRole;
+	}
+
+	private async storeWorkflow(workflow: object, user: User) {
+		await getConnection().transaction(async (transactionManager) => {
+			const newWorkflow = new WorkflowEntity();
+
+			Object.assign(newWorkflow, workflow);
+
+			const savedWorkflow = await transactionManager.save<WorkflowEntity>(newWorkflow);
+
+			const newSharedWorkflow = new SharedWorkflow();
+
+			Object.assign(newSharedWorkflow, {
+				workflow: savedWorkflow,
+				user,
+				role: this.ownerWorkflowRole,
+			});
+
+			await transactionManager.save<SharedWorkflow>(newSharedWorkflow);
+		});
+	}
+
+	private async getOwner() {
+		const ownerGlobalRole = await Db.collections.Role!.findOne({
+			where: { name: 'owner', scope: 'global' },
+		});
+
+		const owner = await Db.collections.User!.findOne({ globalRole: ownerGlobalRole });
+
+		if (!owner) {
+			throw new Error(`No owner found. ${FIX_INSTRUCTION}`);
+		}
+
+		return owner;
+	}
+
+	private async getAssignee(id: string) {
+		const user = await Db.collections.User!.findOne(id);
+
+		if (!user) {
+			throw new Error(`Failed to find user with ID ${id}. Are you sure this user exists?`);
```

Comment:
let's keep it informative 😉 
```suggestion
			throw new Error(`Failed to find user with ID ${id}`);
```

Reply:
Then remove also the second sentence in this message? 'File does not seem to contain workflows. Make sure the workflows are contained in an array.'

Reply:
I think there it is helpful to explain the format, since it might not be as obvious as in the user error